src/main/java/Application.java

```diff
@@ -0,0 +1,88 @@
+package src.main.java;
+
+import java.util.Scanner;
+
+public class Application {
+	private ATM atm = new ATM(this);
+	private boolean exit = false;
+	private Scanner scan = new Scanner(System.in);
+	
+	public Application() {
+		StartMenuRunning();
+	}
+	
+	public void StartMenuRunning() {
+		while(!this.exit) {
+			Display.Menu();
+			String name = scan.nextLine();
+			ATMaccount account = atm.getAccountUser(name);
+			if(account != null) runningAccount(account);
+		}
+		scan.close();
+		System.out.println("End Of the Session");
+	}
+	
+	public void runningAccount(ATMaccount account) {
+		System.out.println("Welcome " + account.getUser().getName());
+		while(!account.isClosed()) {
+			Display.accountMenu(account);
+			int choice = scan.nextInt();
+			atmWork(account, choice);
+		}
+		System.out.println("End Of the Session of " + account.getUser().getName());
+		scan.nextLine();
+		StartMenuRunning();
+
+	}
+	
+	public void exit() {
+		this.exit= true;
+	}
+	
+	private void atmWork(ATMaccount atm,int choice) {
+		switch(choice) {
+		case 1:
+			System.out.println("Balance of the account");
+			System.out.println(atm.CheckBalance());
+			break;
+		case 2:
+			System.out.println("Give money to deposit");
+			float moneyToDeposit = scan.nextFloat();
+			atm.DepositMoney(moneyToDeposit);
+			System.out.println("Balance of the account");
+			System.out.println(atm.CheckBalance());
+			break;
+		case 3:
+			System.out.println("Give money to deposit");
+			float moneyToDraw = scan.nextFloat();
+			atm.WithdrawMoney(moneyToDraw);
+			System.out.println("Balance of the account");
+			System.out.println(atm.CheckBalance());
+			break;
+		case 4:
+			atm.close();
+			break;
+		default:
+			ErrorHandling.InvalidOption();
+		}
+		
+	}
+	public ATMaccount makeAccount(String name,ATM account) {
+		System.out.println("Do you want to make an account(0/1)");
+		int choice = scan.nextInt();
+		scan.nextLine(); // Consume the newline left-over
+		if(choice == 0) {
+			return null;
+		}
+		else if(choice == 1) {
+			User user = new User(name);
+			account.addAccount(user.getAccount(), name);
+			return user.getAccount();
+		}
+		else {
+			ErrorHandling.InvalidOption();
+			return null;
+		}
```

Comment:
This is a good example where the succession of if -> else if -> else is verbose and the logic can be improved. This block of code can be replaced with this more clear and concise code snippet:

```
if (choice == 0) {
	return null;
}

if (choice == 1) {
	var user = new User(name);
	account.addAccount(user.getAccount(), name);
	return user.getAccount();
}

ErrorHandling.InvalidOption();
return null;
```

Reply:
yeah that's right, is there a reason why it is better like this ? (performance reason maybe?)

Reply:
Rarely it's the case about performance. Keep in mind that the compiler will many times optimize your code, so the byte code that your code gets compiled to might look similar or even be the same for the 2 options. My suggestion is better in terms of verbosity and code readability. As a developer you should always keep in mind that your code will be read by many other developers so you would ideally want to make it as short, clear and concise as possible to improve code cleanliness and maintainability.

Reply:
ok thx